langchain/document_loaders/base.py

```diff
@@ -1,9 +1,10 @@
 """Abstract interface for document loader implementations."""
-
+import abc
 from abc import ABC, abstractmethod
-from typing import Iterable, List, Optional
+from typing import Iterable, Iterator, List, Optional
 
-from langchain.docstore.document import Document
+from langchain.document_loaders.blob_loaders import Blob
```

Comment:
A little bit -- I can move the Blob to the base in a separate PR (will be a breaking change) since it's already there. 

We'll need to come up with a convention about how to namespace and also how to decouple the namespace from the implementation so we can do internal refactoring without worrying about breaking changes.